### ISSUE TITLE
PGO: seeing nonsensical counts and corrupted type handles

### ISSUE THREAD

**Issue by AndyAyersMS** [dotnet/runtime#47930]
See notes: https://github.com/dotnet/runtime/pull/47790#issuecomment-773515226

Suspect the copying done by `getPgoInstrumentationResultsInstance` is still off.

cc @davidwrighton

**Comment by AndyAyersMS** [comment]
This fixes things for me (at least counts look good) ... not sure if it's the right fix overall:
```diff
-        volatile size_t*pSrc = (volatile size_t*)found->header.GetData();
+        volatile size_t*pSrc = (volatile size_t*)(found->header.GetData() + found->header.countsOffset);
```

**Comment by AndyAyersMS** [comment]
Added the above fix to #47959.
